fly/commands/clear_resource_cache.go

```diff
@@ -0,0 +1,65 @@
+package commands
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/vito/go-interact/interact"
+
+	"github.com/concourse/concourse/atc"
+	"github.com/concourse/concourse/fly/commands/internal/flaghelpers"
+	"github.com/concourse/concourse/fly/rc"
+)
+
+type ClearResourceCacheCommand struct {
+	Resource 	 flaghelpers.ResourceFlag `short:"r" long:"resource" required:"true" value-name:"PIPELINE/RESOURCE" description:"Name of a resource to clear cache"`
+	Version      *atc.Version             `short:"v" long:"version"                  value-name:"VERSION"           description:"Version of the resource to check from, e.g. digest:sha256@..., in case a version is not specified the command will delete all the resource caches for that resource"`
+}
+
+func (command *ClearResourceCacheCommand) Execute(args []string) error {
+
+	target, err := rc.LoadTarget(Fly.Target, Fly.Verbose)
+	if err != nil {
+		return err
+	}
+
+	err = target.Validate()
+	if err != nil {
+		return err
+	}
+
+	var version atc.Version
+	if command.Version != nil {
```

Comment:
Since we have a `command.Version != nil` check down below on line 38, we can actually combine this one with the one below. That way we don't need two separate `if command.Version != nil` statements!